### DISCUSSION TITLE
Small application for automatic SVG populating and rule generating

### DISCUSSION THREAD

**Discussion by osfog** [ExperienceLovelace/ha-floorplan#378] [Ideas]
Hi,

I wrote a little helper application for HA  Floorplan. It helps you to populate a floorplan SVG with symbols for the entities you want to have as well as generate the rules segment.

I came from a situation where i had no floor plan but quite a few entities that i wanted to place. To create this and know that I in the future very well might want to change the symbols used or the way they behaved resulted me in taking the time to develop a script that copies a named SVG element for each relevant entity 

Currently you can filter on a combination of: 
- entity type, 
- entity attribute value
- entity familiar name containing a substring

These are pretty arbitrary and could be generalized better but it allowed me to filter out the entities that were relevant in my situation

**Input**:

- SVG File
- yaml file with info on how to filter out relevant entities and the rules that should apply for these 
- URL and access token for the home assistant server

**Output** 
- Updated SVG file 
- yaml file with the HA Floorplan rules 

Happy to make it available if it would be of use to anyone.

*Edit:*
https://github.com/osfog/hafloorplangen

**Comment by exetico** [comment]
Sounds cool. I'd suggest you to share it here, so other people don't have to "request it". You're also welcome to create a repo, and share the link in the discussion, too 😄 .

**Reply by osfog** [reply]
Done: can be found here 
https://github.com/osfog/hafloorplangen

**Reply by Trzinka** [reply]
# Can you please attach here/or on your github page some screenshots so we can see what and how!
